src/services/etherspot.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import {
+  Sdk as EtherspotSdk,
+  NetworkNames,
+  Account as EtherspotAccount,
+  Accounts as EtherspotAccounts,
+  EnvNames,
+} from 'etherspot';
+
+// utils
+import { reportErrorLog } from 'utils/common';
+
+// config
+import { getEnv } from 'configs/envConfig';
+
+
+class EtherspotService {
+  sdk: EtherspotSdk;
+  subscription: any;
+
+  async init(privateKey: string) {
+    const networkName = getEnv().NETWORK_PROVIDER === 'homestead'
+      ? NetworkNames.Mainnet
+      : NetworkNames.Kovan;
+
+    const envName = getEnv().NETWORK_PROVIDER === 'homestead'
+      ? EnvNames.MainNets
+      : EnvNames.TestNets;
+
+    this.sdk = new EtherspotSdk(privateKey, { env: envName, networkName });
+
+    await this.sdk.computeContractAccount({ sync: true }).catch((error) => {
+      reportErrorLog('EtherspotService init computeContractAccount failed', { error });
+    });
+  }
+
+  getAccounts(): Promise<?EtherspotAccount[]> {
+    return this.sdk.getConnectedAccounts()
+      .then(({ items }: EtherspotAccounts) => items)
+      .catch((error) => {
+        reportErrorLog('EtherspotService getAccounts -> getConnectedAccounts failed', { error });
+        return null;
+      });
+  }
+
+  async logout() {
+    if (!this.sdk) return; // not initialized, nothing to do
```

Comment:
Just to let you know i'm not suggesting anything here because of the nature of the function 😂 orrr do you think `logout` should return _something_? Like true / false? Is it any help to us?

Reply:
Not sure, at this moment I don't think we care about the output 🤔  Added `void` return.

Reply:
"I don't think we care about the output" - good enough for me!